Userland/Libraries/LibGL/SoftwareRasterizer.cpp

```diff
@@ -278,10 +278,29 @@ static void rasterize_triangle(const RasterizerOptions& options, Gfx::Bitmap& re
                             pass = z >= *depth;
                             break;
                         case GL_NOTEQUAL:
+#ifdef __SSE__
                             pass = z != *depth;
+#else
+                            pass = __builtin_bit_cast(u32, z) != __builtin_bit_cast(u32, *depth);
```

Comment:
Instead, let's redefine `AK/BitCast.h`'s version of `bit_cast` in terms of this intrinsic, please also guard it with `#if __has_builtin(__builtin_bit_cast)` :)

Reply:
I proposed that, with the thinking that our bit_cast might not always produce the most optimal code the compiler can, due to usign a __builtin_memcpy, although after looking at 
https://godbolt.org/z/orWxnTjvn
both seem to be equivalent, my bad....

Reply:
It's alright to use the intrinsic if the compiler has it, but let's put that ugliness in the implementation of `bit_cast()`.

Reply:
> Instead, let's redefine `AK/BitCast.h`'s version of `bit_cast` in terms of this intrinsic, please also guard it with `#if __has_builtin(__builtin_bit_cast)` :)

Sure!